controllers/ciresource_fsm.go

```diff
@@ -107,6 +108,12 @@ func (f *CIResourceFSM) handleStateProvisioningWait(context CIResourceFSMContext
 
 	isReady, resource, err := context.Provider.AcquireCompleted(context.CIResource.Status.ResourceId)
 	if err != nil {
+		// if provisioning error is permanent stop trying
+		// go back and provision another resource
+		if errors.As(err, &providers.ResourceNotFoundError{}) {
+			f.logger.Info("reacquiring, reasource dissapeared", "Id", context.CIResource.Status.ResourceId, "Error", err.Error())
```

Comment:
In Equinix the failures happen in less than 5 minutes.

But the main point it this: if we release the node while it is in provisioning wait, won't it simply finish to provision, end up in use and then release without an error?

Reply:
> In Equinix the failures happen in less than 5 minutes.
Ack, shouldn't be a problem so

> But the main point it this: if we release the node while it is in provisioning wait, won't it simply finish to provision, end up in use and then release without an error?
yes it will, just wasted for 5 hours doing nothing, but based on the 5 minute thing sounds like its unlikely

Reply:
I looked at the ofcir server code and now I understand, the release command requires the CIR to be in "StateInUse".
I think this could be fixed by adding more retries to curl in the ofcir-release-commands